packages/fuels-macros/src/lib.rs

```diff
@@ -76,3 +81,30 @@ pub fn setup_contract_test(input: TokenStream) -> TokenStream {
         .unwrap_or_else(|e| e.to_compile_error())
         .into()
 }
+
+#[proc_macro_derive(Parameterize, attributes(FuelsTypesPath))]
+pub fn parameterize(stream: TokenStream) -> TokenStream {
+    let input = parse_macro_input!(stream as DeriveInput);
+
+    generate_parameterize_impl(input)
+        .unwrap_or_else(|e| e.to_compile_error())
+        .into()
+}
+
+#[proc_macro_derive(Tokenizable)]
+pub fn tokenizable(stream: TokenStream) -> TokenStream {
+    let input = parse_macro_input!(stream as DeriveInput);
+
+    generate_tokenizable_impl(input)
+        .unwrap_or_else(|e| e.to_compile_error())
+        .into()
+}
+
+#[proc_macro_derive(TryFrom)]
+pub fn try_from(stream: TokenStream) -> TokenStream {
+    let input = parse_macro_input!(stream as DeriveInput);
+
+    generate_try_from_impl(input)
+        .unwrap_or_else(|e| e.to_compile_error())
+        .into()
+}
```

Comment:
This whole thing is a work of art. Excellent work breaking down the layers in a very readable/maintainable way!

Reply:
@segfault-magnet's refactors after the initial implementation are always amazing!